doc/generated_testcases_v2.yaml

```diff
@@ -0,0 +1,221 @@
+# Example generators.yaml
+
+# The solution is used to generate a .ans for each generated .in which doesn't
+# yet have a corresponding .ans. If there are generators that don't write a .ans
+# file themselves, this must be specified.
+# This should read the input from stdin and write to stdout.
+#
+# Absolute paths are taken relative to the problem root, relative paths are
+# taken relative to the generators directory.
+#
+# TOOLING: may pick a default if not specified, but should raise a warning.
+solution: /submissions/accepted/sol.py
+
+# The visualizer is used when no suitable image was generated already.
+# This should read {name}.in and/or {name}.ans from the current working
+# directory, and write {name}.ext for an extension in:
+# .png, .jpg, jpeg, .svg
+#
+# TOOLING: may provide a flag to make running this optional, as it can be slow
+# and usually isn't required.
+visualizer: /visualizers/asy.py {name}
+
+# Optionally, a salt for generating the {seed} variables. Will be prepended to
+# the command being run.
+random_salt: abcd
+
+# We support three types of generators:
+# - Standalone files, like generators/a.cpp, generators/b.py, ..., which will
+#   be compiled and run the same way as submissions.
+# - Directories, like 
+#   - generators/gen/tree.cpp
+#   - generators/gen/lib.h
+#   This will be compiled and run the same was as directory validators. Build
+#   and run scripts may be used.
+# - 'implicit' generators whose dependencies are specified in the generators:
+#   key below. The dependencies may refer to any files relative to generators/.
+#   The generator will be build and run as if they formed a separate directory.
+#   E.g. the first example would be equivalent to
+#   - generators/tree/lib.py
+#   - generators/tree/tree.py
+#
+# When a generator is used as a command to generate a .in file, we first check
+# if the name is a key in the generators: dictionary below. If that is the case,
+# the corresponding generator is used. If not, we will use the file/directory
+# generator in the generator/ directory directly.
+generators:
+# A generator that depends on two files, lib.py and tree.py, directly in the
+# generators directory.
+  tree:
+    - lib.py
+    - tree.py
+# Another generator that also depends on the same lib.py.
+  graph:
+    - lib.py
+    - graph.py
+# This also works for other languages.
+  a:
+    - a.h
+    - a.cpp
+# Single-file generators may be specified, but can also be referred to as
+# b.cpp directly.
+  b:
+    - b.cpp
+# It is allowed, but not required, to explicitly list single-file generators
+# as well. It is allowed to reuse the same name, but introducing a new name is
+# also fine.
+  c.py:
+    - c.py
+    - lib.py
+
+data:
```

Comment:
For later documentation purposes, might be nice to specify that `data:` here is a keyword to specify what testcases can be generated, not an explicit reference to the `/data` directory in a problem package.